pkg/serializer/internal/metrics/iterable_series.go

```diff
@@ -500,17 +500,35 @@ func (pb *PayloadsBuilder) finishPayload() error {
 // FIXME(maxime): to be removed when v2 endpoints are available
 func (series *IterableSeries) MarshalJSON() ([]byte, error) {
 	// use an alias to avoid infinite recursion while serializing a Series
-	type SeriesAlias Series
+	type SeriesAlias struct {
+		*metrics.Serie
+		Metadata map[string]interface{} `json:"metadata,omitempty"`
```

Comment:
Would it make sense to define a real struct for metadata instead of using `map[string]any`? All values are numeric, it should be possible to do this without the extra allocations.

Reply:
Agreed, makes sense 👍